flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/component/DispatcherResourceManagerComponent.java

```diff
@@ -86,6 +88,9 @@
         this.dispatcherLeaderRetrievalService = dispatcherLeaderRetrievalService;
         this.resourceManagerRetrievalService = resourceManagerRetrievalService;
         this.webMonitorEndpoint = webMonitorEndpoint;
+        if (webMonitorEndpoint instanceof WebMonitorEndpoint) {
```

Comment:
TBH, This change is not very good :)

I think we could pass the `webInterfaceUrl` to `KubernetesResourceManagerDriver` and then parse the rest port. You could find the similar implementation in the `YarnResourceManager#registerApplicationMaster`